ansible/action_plugins/agnosticd_user_info.py

```diff
@@ -53,9 +65,15 @@ def run(self, tmp=None, task_vars=None):
                     )
                 )
             )
-            fh = open(os.path.join(output_dir, 'user-info.yaml'), 'a')
-            fh.write('- ' + json.dumps(self._task.args['msg']) + "\n")
-            fh.close()
+            if msg:
+                fh = open(os.path.join(output_dir, 'user-info.yaml'), 'a')
+                fh.write('- ' + json.dumps(msg) + "\n")
+                fh.close()
+            if data:
+                fh = open(os.path.join(output_dir, 'user-data.yaml'), 'a')
+                for k, v in data.items():
+                    fh.write("{0}: {1}\n".format(k, json.dumps(v)))
```

Comment:
I'm wondering  what would happen if the same key is written 2 times ?
There would be duplicates in the file and that would just be a warning ?
Can't we just use `json.dump(data)` directly ?

```python
import json 
import os
try:
    fh = open(os.path.join(output_dir, 'user-data.yaml'), 'r')
    cur_data=json.loads(fh.read())
    cur_data.update(data)
    fh = open(os.path.join(output_dir, 'user-data.yaml'), 'w')
    json.dump(cur_data, fh)
except:
    fh = open(os.path.join(output_dir, 'user-data.yaml'), 'w')
    json.dump(data, fh)
```

I used `try` here and you would need to handle errors better than that, but it's just to illustrate. Wdty?

Reply:
It seemed simpler to me to not have to read the file in, just taking advantage of JSON documents being valid inside of YAML documents. It's not a strong opinion, but it is more lines of code and more cpu/memory usage to read it in, update, and write it back out.

If someone set the same data twice then... well, that would be a silly thing to do. Undefined behavior. Probably the YAML parser would override the first setting with the second, but no promises.

Reply:
I was thinking when developing from laptop, could that fail because of consecutive runs ?

Reply:
OK i see the initialization now. All good